### ISSUE TITLE
Console Provider Scope

### ISSUE THREAD

**Issue by ericrini** [aspnet/Logging#793]
I am looking at this article, which suggests that ILogger.BeginScope can be given an IEnumerable<KeyValuePair<string, object>> and will treat each item in the enumerable as if it were a separate scope entry.

https://nblumhardt.com/2016/11/ilogger-beginscope/

In the Console based implementation of ILogger, I am looking at line 248 and I am not seeing it make any special case handling for IEnumerable.

https://github.com/aspnet/Logging/blob/000418aca3936a317309352c0a627c1b512f2c95/src/Microsoft.Extensions.Logging.Console/ConsoleLogger.cs

Whenever I set the scope to an IEnumerable I get this output.

```shell
crit: Test.Service[0]
      => System.Collections.Generic.Dictionary`2[System.String,System.Object]
      EventLevel.Critical
```

So basically the bug is that  I expect to get this output.

```shell
crit: Test.Service[0]
      Key 1 => Value 1
      Key 2 => Value 2
      Key 3 => Value 3
      EventLevel.Critical
```

If someone in the know can confirm this is a correct expectation, I'm happy to submit a pull request to fix it. Thanks.

**Comment by pakrym** [comment]
You have to implement ToString on your scope object (https://github.com/aspnet/Hosting/blob/5543d29af1e6b4f2f5dfa8822db12b666f33d1b5/src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs#L95) to display it correctly in console logger.

`IEnumerable<KeyValuePair<string, object>>` is mostly used by structured loggers, text based loggers would just call ToString.
